### PR TITLE
VIH-10236 RemoveUserAPIClient

### PR DESCRIPTION
**Pull request by muralikaturi**
### Change description ###
Removed UserApi.Client code references to retrieve the user profile as this can be populated from the ClaimsPrincipal

### CODE REVIEW

File: VideoWeb/VideoWeb.UnitTests/Controllers/ProfileController/GetProfileByUsernameTests.cs [hunk 1]
```diff
@@ -43,54 +35,54 @@ public void Setup()
             };
 
             _mocker.Mock<IMapperFactory>().Setup(x => x.Get<UserProfile, UserProfileResponse>()).Returns(_mocker.Create<UserProfileToUserProfileResponseMapper>());
-            var parameters = new ParameterBuilder(_mocker).AddObject(new DictionaryUserCache()).Build();
-            _sut = _mocker.Create<ProfilesController>(parameters);
+            //var parameters = new ParameterBuilder(_mocker).AddObject(new U()).Build();
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb.UnitTests/Controllers/ProfileController/GetProfileByUsernameTests.cs:38]: please remove all the commented code

File: VideoWeb/VideoWeb.UnitTests/Hub/EventHubBaseTests.cs [hunk 2]
```diff
@@ -57,16 +59,16 @@ public void Setup()
             HubCallerContextMock.Setup(x => x.ConnectionId).Returns(Guid.NewGuid().ToString());
             HubCallerContextMock.Setup(x => x.UserIdentifier).Returns(Claims.Identity.Name);
 
-            UserProfileServiceMock.Setup(x => x.GetObfuscatedUsernameAsync(It.IsAny<string>()))
-                .ReturnsAsync("o**** f*****");
+            UserProfileServiceMock.Setup(x => x.GetObfuscatedUsername(It.IsAny<string>()))
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb.UnitTests/Hub/EventHubBaseTests.cs:62]: I'm pretty sure the only reason we have `GetObfuscatedUsername` is for logging purposes. I would suggest we get rid of it if that's still the case

File: VideoWeb/VideoWeb.UnitTests/Mappings/UserProfileToUserProfileResponseMapperTests.cs [hunk 3]
```diff
@@ -33,15 +33,15 @@ public void Should_map_to_user_role(string role, Role expectedRole)
...
         }
 
-        [Test]
-        public void Should_throw_exception_when_role_is_unsupported()
-        {
-            Action action = () => _sut.Map(new UserProfile
-            {
-                UserRole = "Random"
-            });
+        //[Test]
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb.UnitTests/Mappings/UserProfileToUserProfileResponseMapperTests.cs:36]: Safe to remove?

File: VideoWeb/VideoWeb.Common/Caching/DistributedUserClaimsCache.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    public interface IUserClaimsCache
+    {
+        Task<List<Claim>> SetAsync(string key, List<Claim> userClaims);
+        Task<List<Claim>> GetAsync(string key);
+        Task ClearFromCache(string key);
+    }
+
+    public class DistributedUserClaimsCache : RedisCacheBase<string, List<Claim>>, IUserClaimsCache
+    {
+        private readonly string _entryPrefix = "userclaims_";
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb.Common/Caching/DistributedUserClaimsCache.cs:18]: Good call!

**Review by shaed-parkar — COMMENTED** [review]
Nice work! Couple of points worth thinking about else I think this is ready for QA

File: VideoWeb/VideoWeb.UnitTests/Hub/EventHubBaseTests.cs (hunk 2 above)
Comment (muralikaturi) [reply VideoWeb/VideoWeb.UnitTests/Hub/EventHubBaseTests.cs:62]: Not sure. We will have to come back to this as part of the refactoring of the event hub code. It appears there are some API calls made without caching implementation.

File: VideoWeb/VideoWeb.UnitTests/Controllers/ProfileController/GetProfileByUsernameTests.cs (hunk 1 above)
Comment (muralikaturi) [reply VideoWeb/VideoWeb.UnitTests/Controllers/ProfileController/GetProfileByUsernameTests.cs]: done

File: VideoWeb/VideoWeb.UnitTests/Mappings/UserProfileToUserProfileResponseMapperTests.cs (hunk 3 above)
Comment (muralikaturi) [reply VideoWeb/VideoWeb.UnitTests/Mappings/UserProfileToUserProfileResponseMapperTests.cs]: done

**Review by shaed-parkar — APPROVED** [review]

**Review by komaldabhi1509 — APPROVED** [review]

**Pull request merged**